test/BKTClient.spec.ts

```diff
@@ -0,0 +1,817 @@
+                  evaluations: [buildEvaluation(value)],
+                },
+                userEvaluationsId: 'user_evaluation_id_value',
+              }),
+            )
+          }),
+        )
+
+        await initializeBKTClient(config, toBKTUser(user1), 1000)
+
+        const client = getBKTClient()
+
+        assert(client !== null)
+
+        expect(client.numberVariation('feature_id_value', defaultValue)).toBe(
+          actual,
+        )
+      },
+    )
+
+    test('returns default value if feature is not found', async () => {
+      server = setupServerAndListen(
+        rest.post<
+          GetEvaluationsRequest,
+          Record<string, never>,
+          GetEvaluationsResponse
+        >('https://api.bucketeer.io/get_evaluations', (_req, res, ctx) => {
+          return res.once(
+            ctx.status(200),
+            ctx.json({
+              evaluations: user1Evaluations,
+              userEvaluationsId: 'user_evaluation_id_value',
+            }),
+          )
+        }),
+      )
+
+      await initializeBKTClient(config, toBKTUser(user1), 1000)
+
+      const client = getBKTClient()
+
+      assert(client !== null)
+
+      expect(client.numberVariation('feature_id_value', 99)).toBe(99)
+    })
+  })
+
+  suite('booleanVariation', () => {
+    test.each([
+      ['true', false, true],
+      ['false', true, false],
+      ['true', true, true],
+      ['TRUE', false, true],
+      ['truea', false, false],
+      ['not bool', false, false],
+      ['not bool', true, true],
+      ['1', false, false],
+      ['{}', false, false],
+    ])(
+      'value=%s, default=%s, actual=%s',
+      async (value: string, defaultValue: boolean, actual: boolean) => {
+        server = setupServerAndListen(
+          rest.post<
+            GetEvaluationsRequest,
+            Record<string, never>,
+            GetEvaluationsResponse
+          >('https://api.bucketeer.io/get_evaluations', (_req, res, ctx) => {
+            return res.once(
+              ctx.status(200),
+              ctx.json({
+                evaluations: {
+                  ...user1Evaluations,
+                  evaluations: [buildEvaluation(value)],
+                },
+                userEvaluationsId: 'user_evaluation_id_value',
+              }),
+            )
+          }),
+        )
+
+        await initializeBKTClient(config, toBKTUser(user1), 1000)
+
+        const client = getBKTClient()
+
+        assert(client !== null)
+
+        expect(client.booleanVariation('feature_id_value', defaultValue)).toBe(
+          actual,
+        )
+      },
+    )
+
+    // cases for defaultValue is covered in the above test
+  })
+
+  suite('jsonVariation', () => {
+    const JSON_VALUE = '{"key": "value"}'
+
+    test.each([
+      [JSON_VALUE, {}, JSON.parse(JSON_VALUE)],
+      ['true', JSON.parse(JSON_VALUE), true],
+      ['true', {}, true],
+      ['not bool', {}, {}],
+      ['1', {}, 1],
+      ['{}', {}, {}],
```

Comment:
do we need `jsonVariation`/`numberVariation`/`booleanVariation` in js SDK?

As you can see from the test case, `JSON.parse()` can parse boolean string and number string. so the behavior here is not consistent with Android SDK.

(also, previous bucketeer-web-sdk only has `stringVariation()` function)

Reply:
I want to unify the interfaces as much as I can. So, for now, let's keep it as it is.